codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/serde/SerializeStructGenerator.kt

```diff
@@ -463,15 +480,26 @@ open class SerializeStructGenerator(
      * input.fooBlobMap.forEach { (key, value) -> entry(key, value.encodeBase64String()) }
      * ```
      */
-    private fun renderBlobEntry(keyShape: Shape, nestingLevel: Int, listMemberName: String) {
+    private fun renderBlobEntry(keyShape: Shape, nestingLevel: Int, listMemberName: String, isSparse: Boolean) {
         val containerName = if (nestingLevel == 0) "input." else ""
         val (keyName, valueName) = keyValueNames(nestingLevel)
         val keyValue = keyValue(keyShape, keyName)
 
-        writer.write(
-            "$containerName$listMemberName.forEach { ($keyName, $valueName) -> entry($keyValue, $valueName.#T()) }",
-            RuntimeTypes.Core.Text.Encoding.encodeBase64String,
-        )
+        when (isSparse) {
+            true -> {
+                writer.withBlock("$containerName$listMemberName.forEach { ($keyName, $valueName) ->", "}") {
+                    writer.withBlock("if (value != null) {", "} else entry(key, null as String?)") {
```

Comment:
question: Here we are assuming the value will always be a string (`null as String?`), I think that works because serializers will handle a `null` entry of any type the same way?

Reply:
>question: Here we are assuming the value will always be a string (null as String?), I think that works because serializers will handle a null entry of any type the same way?

Yes that's right, there should be a way to use the correct value type but it's not simple and the results would be the same

Reply:
Actually that's only true for the `JsonSerializer` because of the `XmlSerializer` we need to use `as String?`
https://github.com/smithy-lang/smithy-kotlin/blob/7d2012af6077ee1619b6ff66f73edba53081f3e0/runtime/serde/serde-xml/common/src/aws/smithy/kotlin/runtime/serde/xml/XmlSerializer.kt#L231-L263
`null.toString()` returns the string `"null"` according to the docs